### PR TITLE
fix(firestore-bigquery-change-tracker): added additional checks on table creation

### PR DESCRIPTION
**Pull request by dackers86**
- [x] Table creation errors are now caught, allowing records to still be written if creating through a race condition.
  - [x] Additional database checks added before writing data to BigQuery

fixes: https://github.com/firebase/extensions/issues/869

### CODE REVIEW

**Comment by dackers86** [comment]
### Testing


image

File: firestore-bigquery-export/firestore-bigquery-change-tracker/src/bigquery/index.ts [hunk 1]
```diff
@@ -148,6 +148,20 @@ export class FirestoreBigQueryEventHistoryTracker
     return isRetryable;
   }
 
+  private async waitForInitialization(dataset, table) {
```
Comment (Ehesp) [inline firestore-bigquery-export/firestore-bigquery-change-tracker/src/bigquery/index.ts:151]: Probably worth adding a code comment on this to explain the reasoning for future reference.

Comment (dackers86) [reply firestore-bigquery-export/firestore-bigquery-change-tracker/src/bigquery/index.ts:156]: Comment added.

File: firestore-bigquery-export/firestore-bigquery-change-tracker/src/bigquery/index.ts [hunk 2]
```diff
@@ -148,6 +148,25 @@ export class FirestoreBigQueryEventHistoryTracker
...
+   * continually re-checks until the referenced dataset and table become available.
+   */
+  private async waitForInitialization(dataset, table) {
+    return new Promise((resolve) => {
+      let handle = setTimeout(async () => {
+        const [datasetExists] = await dataset.exists();
+        const [tableExists] = await table.exists();
+
+        if (datasetExists && tableExists) {
+          clearTimeout(handle);
```
Comment (Salakar) [inline firestore-bigquery-export/firestore-bigquery-change-tracker/src/bigquery/index.ts:163]:
```suggestion
      let handle = setInterval(async () => {
        const [datasetExists] = await dataset.exists();
        const [tableExists] = await table.exists();

        if (datasetExists && tableExists) {
          clearInterval(handle);
```

**Pull request merged**